File: APIBanHangReactNative/models/model_address.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const address = new Schema({
    fullName: { type: String, required: true },
    province: { type: String, required: true },
    district: { type: String, required: true },
    commune: { type: String, required: true },
    receivingAddress: { type: String, required: true },
    phone: { type: Number, required: true },
    id_user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
}, {
    timestamps: true,
});

module.exports = mongoose.model('address', address);

File: app_banhang2/lib/services/models/model_address.dart
class ModelAddress {
  final String id;
  final String fullName;
  final String province;
  final String district;
  final String commune;
  final String receivingAddress;
  final String phone;
  final String idUser;
  final bool isDefault;

  ModelAddress({
    required this.id,
    required this.fullName,
    required this.province,
    required this.district,
    required this.commune,
    required this.receivingAddress,
    required this.phone,
    required this.idUser,
    this.isDefault = false,
  });

  factory ModelAddress.fromJson(Map<String, dynamic> json) {
    return ModelAddress(
      id: json['_id'] ?? '',
      fullName: json['fullName'] ?? '',
      province: json['province'] ?? '',
      district: json['district'] ?? '',
      commune: json['commune'] ?? '',
      receivingAddress: json['receivingAddress'] ?? '',
      phone: json['phone']?.toString() ?? '',
      idUser: json['id_user'] ?? '',
      isDefault: json['chooseDefault'] ?? false,
    );
  }
}